cyclonedx/model/component.py

```diff
@@ -766,7 +766,7 @@ def mime_type(self, mime_type: Optional[str]) -> None:
         self._mime_type = mime_type
 
     @property
-    def bom_ref(self) -> Optional[str]:
+    def bom_ref(self) -> str:
```

Comment:
seeing this change makes me wonder, if it might be a goot sime to switch the `bom_ref` from optional strings to 
an non-optional read-only object of a new data type `BomRef`.

the ref/bom-ref properties in CycloneDX are a special kind of string. they are inteded to link elements together.
therefore bom-refs are used, instead of the actual linked elements.
having special data types makes it easy to tell elements apart by the reference.

proposeal: 
have a data class `BomRef`. `BomRef` has a property value of type `Optional[str]`.

Reply:
what d you think, @madpah ?

Reply:
#170 raised and will consider for this release @jkowalleck